web/src/components/product/RegistrationExtension.tsx

```diff
@@ -136,24 +141,18 @@ export default function RegistrationExtension({
           <RegisteredExtensionStatus registrationCode={registrationData.registrationCode} />
         )}
         {!isRegistered && extension.available && !extension.free && (
-          <Form id={`register-form-${extension.id}-${extension.version}`} onSubmit={submit}>
+          <Form id={formId} onSubmit={submit}>
             {/* // TRANSLATORS: input field label */}
-            <FormGroup label={_("Registration code")}>
+            <FormGroup fieldId={inputId} label={_("Registration code")}>
```

Comment:
I've updated the `id` value here to ensure proper association between the label and its corresponding input. This was identified while writing unit tests using [getByLabelText](https://testing-library.com/docs/queries/bylabeltext/), which relies on that association. We're using this method specifically for password fields, since they can't be queried by role because they has no implicit role. Read https://github.com/testing-library/dom-testing-library/issues/567